src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/PolicyCmdletBase.cs

```diff
@@ -106,5 +148,97 @@ protected JToken GetObjectFromParameter(string parameter)
                 ? JToken.FromObject(FileUtilities.DataStore.ReadFileAsText(path))
                 : JToken.FromObject(parameter);
         }
+
+        /// <summary>
+        /// Gets the resource Id from the supplied PowerShell parameters.
+        /// </summary>
+        protected string MakePolicyAssignmentId(string scope, string resourceName)
+        {
+            return ResourceIdUtility.GetResourceId(
```

Comment:
I think there was a bug with this method where it didn't produce the right thing if the assignment was an extension of another resource (i.e. an MG assignment or assignment on an individual resource)

Reply:
You might be thinking about the ResourceIdUtility.GetResourceName(resourceId) method. Our use of that one definitely got broken by MG scopes, but I didn't modify the method itself because it's used all over the place and I didn't want to break other stuff or spread this PR into other cmdlets. I just changed our cmdlet logic to only use it in a safe way. I didn't see any problems with GetResourceId( ... ), and I verified that one at all scope levels. It's pretty trivial: it just glues stuff containing Name to the end of Scope (at least the way we use it).